src/traits.rs

```diff
@@ -871,6 +939,22 @@ macro_rules! prim_int_impl {
                 <$T>::rotate_right(self, n)
             }
 
+            fn arithmetic_shl(self, n: u32) -> Self {
+                ((self as $S) << n) as $T
```

Comment:
Is it defined behaviour in Rust? It would be good to provide link to specs for that as I am not sure that it will be consistent between platforms.